op-service/solabi/util.go

```diff
@@ -0,0 +1,131 @@
+package solabi
+
+import (
+	"bytes"
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"io"
+	"math/big"
+
+	"github.com/ethereum-optimism/optimism/op-node/eth"
+	"github.com/ethereum/go-ethereum/common"
+)
+
+// These are empty padding values. They should be zero'd & not modified at all.
+var (
+	addressEmptyPadding [12]byte = [12]byte{}
+	uint64EmptyPadding  [24]byte = [24]byte{}
+)
+
+func ReadSignature(r io.Reader) ([]byte, error) {
+	sig := make([]byte, 4)
+	_, err := io.ReadFull(r, sig)
+	fmt.Println(sig)
+	return sig, err
+}
+
+func ReadAndValidateSignature(r io.Reader, expectedSignature []byte) ([]byte, error) {
+	sig := make([]byte, 4)
+	if _, err := io.ReadFull(r, sig); err != nil {
+		return nil, err
+	}
+	if !bytes.Equal(sig, expectedSignature) {
+		return nil, errors.New("invalid function signature")
+	}
+	return sig, nil
+}
+
+func ReadHash(r io.Reader) (common.Hash, error) {
+	var h common.Hash
+	_, err := io.ReadFull(r, h[:])
+	return h, err
+}
+
+func ReadEthBytes32(r io.Reader) (eth.Bytes32, error) {
+	var b eth.Bytes32
+	_, err := io.ReadFull(r, b[:])
+	return b, err
+}
+
+func ReadAddress(r io.Reader) (common.Address, error) {
+	var padding, readPadding [12]byte
```

Comment:
Since `padding` is a constant, maybe reuse the `addressEmptyPadding` global constant?

Reply:
nice catch. Will reuse `padding`.